src/reducers/__tests__/subCategorizeAll.ts

```diff
@@ -89,28 +90,42 @@ it('set cursor on new empty thought', () => {
 
 })
 
-it('move all visible and hidden thoughts into a new empty thought after subcategorizeAll', () => {
+it('move all non meta thoughts and only allowed meta thoughts into new empty thought after subCategorizeAll', () => {
+
+  const metaExceptions = SUBCATEGORIZE_ALL_META_EXCEPTIONS.reduce(
+    (acc, meta, index) => (acc + `    - ${meta}${index === (SUBCATEGORIZE_ALL_META_EXCEPTIONS.length - 1) ? '' : '\n'}`)
+  , '')
+
+  const text = `
+  - a
+${metaExceptions}   
+    - =style
+    - =view
+    - c
+    - d
+    - e`
 
   const steps = [
-    newThought('b'),
-    newSubthought('=archive'),
-    newThought('c'),
-    newSubthought('d'),
-    cursorBack,
-    subCategorizeAll
+    importText({
+      text,
+      path: HOME_PATH
+    }),
+    setCursorFirstMatch(['a', 'c']),
+    subCategorizeAll,
   ]
 
-  // run steps through reducer flow
+  // run steps through reducer flow and export as plaintext for readable test
   const stateNew = reducerFlow(steps)(initialState())
-
   const exported = exportContext(stateNew, [HOME_TOKEN], 'text/plain')
 
-  const expectedOutput = `- ${HOME_TOKEN}
-  - b
-    -${' '}
-      - =archive
-      - c
-        - d`
 
-  expect(exported).toEqual(expectedOutput)
-})
+  expect(exported).toBe(`- ${HOME_TOKEN}
+  - a
+    - ${''/* prevent trim_trailing_whitespace */}
+      - =style
+      - =view
+      - c
+      - d
+      - e
+${metaExceptions}`)
+})
```

Comment:
Do you have your editor set up to use `.editorconfig`? It should not be allowing you to save files without final newlines. Unless that's a Windows thing.

Reply:
No, I am using Macbook. I think the `eslint-config-prettier` is turning off some important eslint rules from `eslint-config-raine` and possibly other plugins too. All formatting related rules should be handled by the `prettier` now.

Reply:
I don't think prettier has the equivalent of `insert_final_newline` from [EditorConfig](https://editorconfig.org/). You should install EditorConfig in your editor if you don't have it already.

This isn't related to eslint rules, but I am working with App Sapiens to make sure https://github.com/prettier/eslint-config-prettier is not a problem. It is disabling rules but only ones that Prettier can fix automatically.

Reply:
This issue seems to suggest that Prettier does this automatically: https://github.com/prettier/prettier/issues/2575

However it may depend on your editor settings.

Reply:
> I don't think prettier has the equivalent of `insert_final_newline` from [EditorConfig](https://editorconfig.org/). You ~should~ could install EditorConfig in your editor ~if you don't have it already.~ if it doesn't have a setting to automatically add a newline at the end of a file.

Sure.